### DISCUSSION TITLE
Difference between `ActorRefFrom` and `ActorLogicFrom` type in V5

### DISCUSSION THREAD

**Discussion by bhvngt** [statelyai/xstate#4571] [General] [answered]
While working with v5, I often get confused with [ActorRefFrom](https://github.com/statelyai/xstate/blob/a5c55fae213119f83ccb4813c1cb7f028190086c/packages/core/src/types.ts#L1961) and [ActorLogicFrom](https://github.com/statelyai/xstate/blob/a5c55fae213119f83ccb4813c1cb7f028190086c/packages/core/src/types.ts#L1941) types and their use cases.

It will be great if we can have a documentation for these and other types that extracts specific type info from the machine config.

**Comment by davidkpiano** [comment, marked as answer]
Sure, we will add documentation for that.

In short:
```ts
const actorRef = createActor(actorLogic);
```

The `ActorRef` is the live actor instance that you can send events to. The `ActorLogic` represents the logic of an actor.

E.g. a traffic light actor may have `trafficLightMachine` behavior that tells it how to transition between its red, yellow, green states, but there may be many live actor instances (actor refs) of that `trafficLightMachine` behavior.

**Reply by bhvngt** [reply]
Thanks @davidkpiano for your reply.
